compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -161,9 +160,38 @@ public BallerinaSemanticModel(BLangPackage bLangPackage, CompilerContext context
 
     // Private helper methods for the public APIs above.
 
-    private boolean isSymbolInUserProject(BSymbol symbol, DiagnosticPos cursorPos) {
+    private boolean isSymbolInUserProject(BSymbol symbol, BLangDiagnosticLocation cursorPos) {
+        String packageDetailsString = bLangPackage.packageID.name.value
+                + bLangPackage.packageID.getPackageVersion().value
+                + cursorPos.lineRange().filePath();
+        String symbolPackageDetailsString = symbol.pkgID.name.value
+                + symbol.pkgID.getPackageVersion().value
+                + symbol.pos.lineRange().filePath();
+
+        int value = packageDetailsString.compareTo(symbolPackageDetailsString);
+
+        BLangDiagnosticLocation location = symbol.pos;
+
+        if (value == 0) {
+            // If the package detail strings are same, then compare the start line.
+            if (cursorPos.lineRange().startLine().line() < location.lineRange().startLine().line()) {
```

Comment:
`cursorPos.lineRange().startLine()` and `location.lineRange().startLine()` is called multiple times.
Better to extract that to a variable.